src/ninja.cc

```diff
@@ -185,6 +187,10 @@ struct NinjaMain : public BuildLogUser {
   }
 
   int64_t start_time_millis_;
+
+  static std::string GetWorkingDirectory();
+
+  std::string working_directory = GetWorkingDirectory();
```

Comment:
hm ... why have this here which means it runs at startup although it might not be needed? Am I missing something?

Reply:
so, i was thinking that this would save a little complexity in the NinjaMain::ToolCompilationDatabase* functions, in exchange for a negligible added cost for non-compdb tools.

But i neglected to actually measure that cost until now. 

Based on some quick-&-dirty uses of SDL_GetCurrentTime() at the start & end of NinjaMain::GetWorkingDirectory(), it looks like the cost is somewhere in the ballpark of 13 to 15 microseconds. (I also stepped through the instructions of getcwd() on my system and found that the number of syscalls actually executed is somewhat larger than i expected.)

And while 15µs doesn't seem like much on its own, if we allow small unnecessary time expenditures like this as a _policy_, sooner or later they could add up to something noticeable.

So, I'll go ahead and move that call to GetWorkingDirectory(); out of the NinjaMain constructor and into the compdb functions.